core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -720,5 +720,14 @@ public void loadNext() throws IOException {
 
     @Override
     public long getKeyPrefix() { return current.getKeyPrefix(); }
+
+    private void initializeNumRecords() throws IOException {
+      if (numRecords == 0) {
+        for (UnsafeSorterIterator iter: iterators) {
+          numRecords += iter.getNumRecords();
+        }
+        numRecords += current.getNumRecords();
+      }
+    }
```

Comment:
nit: How about this? (Semantically, `this.current` should be null before `initializeNumRecords` called)
```
    ChainedIterator(Queue<UnsafeSorterIterator> iterators) {
      ...
      this.current = null;
    }

    private void initializeNumRecords() throws IOException {
      if (numRecords == 0) {
        for (UnsafeSorterIterator iter: iterators) {
          numRecords += iter.getNumRecords();
        }
        this.current = iterators.remove();
      }
    }
```

Reply:
Thank you for the suggestion. It make sense. I will do it and test it.